platform-tests/src/test/java/org/junit/platform/commons/util/ReflectionUtilsTests.java

```diff
@@ -886,6 +886,18 @@ private void assertFindMethodByParameterNames(String methodName, Class<?> parame
 		assertThat(optional).contains(method);
 	}
 
+	@Test
+	void findMethodChooseSpecificOverGenericMethod() throws NoSuchMethodException {
+		String methodName = "foo";
+		Class<?> parameterType = Double.class;
+		Class<?> testClass = InterfaceWithSpecificAndGenericDefaultMethodImpl.class;
+
+		Method specificMethod = testClass.getMethod(methodName, parameterType);
+		Optional<Method> optional = findMethod(testClass, methodName, parameterType);
+
+		assertThat(optional).isEqualTo(specificMethod);
```

Comment:
That will always fail since you're comparing an `Optional` to a `Method`.

I think you need to use `assertThat(optional).contains(method)`.

Reply:
Woops! You are right. Oversight from my part. Will immediately address this.